src/nvim/globals.h

```diff
@@ -746,6 +746,9 @@ EXTERN bool enc_utf8 INIT(= false);             /* UTF-8 encoded Unicode */
 EXTERN int enc_latin1like INIT(= TRUE);         /* 'encoding' is latin1 comp. */
 EXTERN int has_mbyte INIT(= 0);                 /* any multi-byte encoding */
 
+// 'encoding' can only be changed in the startup scripts
+EXTERN bool enc_mutable INIT(= true);
```

Comment:
Is there a variable that already tracks whether `VimEnter` was raised? Or, can we name this something more generalized, which describes the phase that has been completed, e.g. `is_init_completed`.